File: cpp_src/toggle_forward/src/toggle_server.cpp
#include <ros/ros.h>
#include <std_srvs/Empty.h>
#include <geometry_msgs/Twist.h>
#include <thread>

bool forward = true;

bool toggleForward(std_srvs::Empty::Request &req, std_srvs::Empty::Response &resp) {
    forward = !forward;
    ROS_INFO("now receiving : %s", forward ? "forward" : "rotate");
    return true;
}

int main(int argc, char ** argv) {
    ros::init(argc, argv, "toggle_server");
    ros::NodeHandle nh;

    ros::ServiceServer server = nh.advertiseService("toggle_forward", &toggleForward);

    ros::Publisher pub = nh.advertise<geometry_msgs::Twist>("turtle1/cmd_vel", 1000);

    auto thread_pub = [=]() {
        ros::Rate rate(2);
        while(ros::ok()) {
            geometry_msgs::Twist msg;
            msg.linear.x = forward? 1.0 : 0.0;
            msg.angular.z = forward ? 0.0 : 1.0;
            pub.publish(msg);
            rate.sleep();
        }
    };

    std::thread th1 = std::thread(thread_pub);

    ros::spin();

    if (th1.joinable()) {
        th1.join();
    }

    return 0;
}

File: pyhon_src/toggle_forward/scripts/toggle_server.py
#! /usr/bin/env python3

import rospy
from geometry_msgs.msg import Twist
from std_srvs.srv import Empty, EmptyResponse

g_forward = True

def toggleForward(req):
    global g_forward
    g_forward = not g_forward

    fwd = "forward" if g_forward else "rotate"

    rospy.loginfo("now receiveing: %s" %fwd)

    return EmptyResponse()

def main():
    global g_forward

    rospy.init_node("toggle_server")

    srv = rospy.Service("toggle_forward", Empty, toggleForward)

    pub = rospy.Publisher("turtle1/cmd_vel", Twist, queue_size=10)
    rate = rospy.Rate(2)
    while not rospy.is_shutdown():
        msg = Twist()
        msg.linear.x = 1.0 if g_forward else 0.0
        msg.angular.z = 0.0 if g_forward else 1.0

        pub.publish(msg)

        rate.sleep()



    rospy.spin()

if __name__ == "__main__":
    main()